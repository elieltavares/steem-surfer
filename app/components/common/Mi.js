import React, { Component } from 'react';

import PropTypes from 'prop-types';

class Mi extends Component {
  static defaultProps = {
    rotatedRight: false
  };

  render() {
    const { icon, rotatedRight } = this.props;

    const className = `mi ${rotatedRight ? 'rotated-right' : ''}`;

    return <i className={className}>{icon}</i>;
  }
}

Mi.defaultProps = {
  rotatedRight: false
};

Mi.propTypes = {
  icon: PropTypes.string.isRequired,
  rotatedRight: PropTypes.bool
};

export default Mi;
