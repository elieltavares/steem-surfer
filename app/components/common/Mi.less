
.mi {
  // font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 1em; /* Preferred icon size */
  display: inline-block;
  line-height: 1;
  text-transform: none;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;

  /* Support for IE. */
  font-feature-settings: 'liga';

  &.mi-2x {
    font-size: 2em;
  }

  &.mi-3x {
    font-size: 3rem;
  }

  &.mi-4x {
    font-size: 4rem;
  }

  &.mi-5x {
    font-size: 5rem;
  }

  &.miRotatedRight {
    -webkit-transform: rotate(90deg);
  }
}