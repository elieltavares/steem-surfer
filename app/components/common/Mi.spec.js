/* eslint-disable */
import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Mi from './Mi';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    icon: 'code'
  };
  const component = shallow(<Mi {...props} />);
  return {
    component,
    props
  };
}

describe('Material icon component', () => {
  it('should display icon', () => {
    const { component, props } = setup();
    expect(component.text()).toBe(props.icon);
  });

  it('should match exact snapshot', () => {
    const { props } = setup();
    const mi = (
      <div>
        <Mi {...props} />
      </div>
    );
    const tree = renderer.create(mi).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
