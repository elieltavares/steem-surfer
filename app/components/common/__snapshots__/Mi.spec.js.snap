// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material icon component should match exact snapshot 1`] = `
<div>
  <i
    className="mi "
  >
    code
  </i>
</div>
`;
